src/Questionnaire/Questionnaire.js

```diff
@@ -0,0 +1,56 @@
+import PropTypes from 'prop-types'
+import React, { useState } from 'react'
+import { useHistory } from 'react-router-dom'
+import { styleData } from '../data/styleData'
+
+Questionnaire.propTypes = {
+  userStyles: PropTypes.arrayOf(PropTypes.number).isRequired,
+  onQuestionnaireEnd: PropTypes.func.isRequired,
+  round: PropTypes.number.isRequired,
+}
+
+export default function Questionnaire({
+  userStyles = [],
+  onQuestionnaireEnd,
+  round,
+}) {
+  const history = useHistory()
+  if (userStyles.length < 3) {
+    history.push('/')
+  }
+
+  const currentTestId = userStyles[round - 1] - 1
+  const questions = styleData[currentTestId]?.questions ?? []
```

Comment:
styleData is a misleading name since it is not used for styling.

Reply:
Renamed it "personalityStyleData"